### ISSUE TITLE
Restructured Syllabus Front End

### ISSUE THREAD

**Issue by SallyMcGrath** [CodeYourFuture/Tech-Ed-Next#2]
If I were to say a module is really a toolbox, with these tools in it, would this make sense to you?

MODULE DIR
title: "Module"
toolbox:
  requirements: what needs to have been done before this module, prep work, prior learning, and setup
  objectives: learning objectives
  backlog: display all coursework from module repo tickets
  product: the module project
 success: how do we know we've completed/passed this module ? 
 delivery: how to deliver this module - coordination, planning, reviewing etc, mainly composed from the assigned content blocks
  timeline:
    - sprint-1
    - sprint-2
    - sprint-3
    - sprint-4

SPRINT DIR

Each sprint would be a mini toolkit, broadly the same. And then to compose the syllabus it's just a matter of plugging in the content links into the config file.

 requirements: what needs to have been done before this sprint, prep work, prior learning, and setup
  objectives: learning objectives for this sprint
  backlog: display all coursework from module repo tickets for this week/sprint
 success: how do we know we've completed this sprint ? 
 delivery: how to deliver this sprint- coordination, planning, reviewing etc, mainly composed from the assigned content blocks
 
Basically the idea is the syllabus is a toolbox filled with resources  -- the syllabus level, the module level, and the sprint level all have the same effective structure. Activities (workshops, guides, small exercises) are all stored in a grab bag that belongs to the course. Resources are composed on to the sprint view. The module view is composed from the sprint view. The course view is composed from the module view.

**Comment by SallyMcGrath** [comment]
![Image](https://user-images.githubusercontent.com/42203406/230392153-c20b1bdd-e115-4b7a-992b-5af42a65dd1e.png)

**Comment by SallyMcGrath** [comment]
dropping some notes on this

- this is in a branch on main syllabus called toolbox-format
- i've done a bunch of revisions and workshopping with various teams
- new structure is simpler and clearer
- the main thrust is "what to do when"

**Comment by SallyMcGrath** [comment]
A module can be generated with a script
It has broadly this structure
<img width="1512" alt="Screenshot 2023-05-04 at 08 34 15" src="https://user-images.githubusercontent.com/42203406/236486314-541ff413-5417-415c-b0d5-2b3dcee767dc.png">

1. prep
2. sprint 1
3. sprint 2 ...
4. ..
5. ..
6. backlog
7. product
8. success

A sprint is structured like this
<img width="1512" alt="Screenshot 2023-05-04 at 08 34 25" src="https://user-images.githubusercontent.com/42203406/236486357-3bcca53d-9261-4357-b17a-4b23b42a7363.png">

1. prep
2. backlog
3. day plan
4. success

The prep work is what must be done before class day. This can be composed from videos, text blocks, github readmes, etc. It’s created from a list of blocks:
<img width="1512" alt="Screenshot 2023-05-04 at 08 34 38" src="https://user-images.githubusercontent.com/42203406/236486396-f985e5ec-6f60-43ca-bad8-365f018392dc.png">

The backlog has all the assignments, essentially replacing Google Classroom. It pulls automatically from the coursework repos on Github and filters on the week label
<img width="1512" alt="Screenshot 2023-04-27 at 18 25 08" src="https://user-images.githubusercontent.com/42203406/236486249-cc021ee9-ab91-4efa-8d13-de2ba46c1d80.png">




The day plan shows the rough plan for the day – energiser, retro, workshops etc. It’s composed from blocks just like the prep view
```
+++
title = 'day-plan'
layout = 'day-plan'
emoji= '📝'
menu_level = ['sprint']
weight = 3
backlog= 'Module-JS3'
backlog_filter= 'Week 1'
[[blocks]]
name="energiser"
[[blocks]]
name="kata"
[[blocks]]
name="morning-break"
[[blocks]]
name="Think Like A Computer"
src="https://github.com/CodeYourFuture/recap-workshops/tree/main/think-like-computer/instructions.md"
[[blocks]]
name="lunch"
+++
```

 

The success view compiles all the learning objectives attached to blocks and shows them as a checkable list

<img width="1511" alt="Screenshot 2023-05-04 at 08 35 14" src="https://user-images.githubusercontent.com/42203406/236486454-50770a36-a931-429a-8208-62ecbc8c10b6.png">


That’s all the views – the others were discarded during iteration/testing/feedback.


These layouts are not final, they are just a rough working look.
